File: FrontEnd/src/app/models/order.ts
export enum OrderStatus {
    PENDING = 'PENDING',
    ACCEPTED = 'ACCEPTED',
    CANCELLED = 'CANCELLED'
}

export interface OrderLine {
    id: number;
    variant: {
        id: number;
        product: {
            id: number;
            name: string;
            images: any[];
        };
        size: string;
        color: string;
    };
    quantity: number;
    price: number;
}

export interface Order {
    id: number;
    user: {
        id: number;
        firstName: string;
        lastName: string;
        email: string;
    };
    orderDate: string;
    status: OrderStatus;
    totalAmount: number;
    trackingCode: string;
    phoneNumber?: string;
    region?: string;
    address?: string;
    postalCode?: string;
    orderLines: OrderLine[];
}

File: src/main/java/com/farah/couture/entities/Order.java
package com.farah.couture.entities;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.time.LocalDateTime;
import java.util.List;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "customer_order") // 'order' is a reserved keyword in SQL
public class Order {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "user_id")
    private User user;

    private LocalDateTime orderDate;

    @Enumerated(EnumType.STRING)
    private OrderStatus status; // PENDING, ACCEPTED, CANCELLED

    private Double totalAmount;

    private String trackingCode; // Unique random code (e.g., ORD-X7A9B2)

    // Delivery Details
    private String phoneNumber;
    private String region;
    private String address;
    private String postalCode;

    @OneToMany(mappedBy = "order", cascade = CascadeType.ALL)
    private List<OrderLine> orderLines;
}
